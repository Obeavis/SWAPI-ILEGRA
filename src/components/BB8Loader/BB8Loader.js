import React from "react";
import "./BB8Loader.scss";

const BB8Loader = () => (
	<div className="artboard">
		<div className="bb8">
			<div className="bb8-body">
				<div className="dot dot-1">
					<div className="line line-1"></div>
					<div className="line line-2"></div>
					<div className="line line-3"></div>
				</div>
				<div className="dot dot-2"></div>
				<div className="circle circle-1"></div>
				<div className="circle circle-2"></div>
				<div className="circle circle-3"></div>
			</div>
			<div className="body-shadow-crop">
				<div className="body-shadow"></div>
			</div>
			<div className="bb8-head">
				<div className="head-bottom">
					<div className="head-side-1"></div>
					<div className="head-side-2"></div>
					<div className="head-bottom-base"></div>
				</div>
				<div className="head-top-crop">
					<div className="head-top"></div>
				</div>
				<div className="lens"></div>
				<div className="freckle"></div>
			</div>
			<div className="speedlines">
				<div className="one tail"></div>
				<div className="two tail"></div>
				<div className="three"></div>
				<div className="four"></div>
				<div className="five tail"></div>
			</div>
			<div className="sparkles">
				<div className="one small pulse-1"></div>
				<div className="two blue small pulse-2"></div>
				<div className="three blue med pulse-3"></div>
				<div className="four orange pulse-2"></div>
				<div className="five orange pulse-1"></div>
				<div className="six blue small pulse"></div>
				<div className="seven blue small pulse"></div>
				<div className="eight small pulse-3"></div>
				<div className="nine pulse"></div>
				<div className="ten orange small-1 pulse"></div>
				<div className="eleven small pulse"></div>
				<div className="twelve small pulse-2"></div>
				<div className="thirteen orange small pulse"></div>
				<div className="fourteen orange med pulse-3"></div>
				<div className="fifteen small pulse-1"></div>
				<div className="sixteen small pulse"></div>
			</div>
			<div className="ground">
				<div className="one">
					<div className="bump move-1"></div>
				</div>
				<div className="two"></div>
				<div className="three">
					<div className="bump move-2"></div>
				</div>
				<div className="four">
					<div className="bump"></div>
				</div>
				<div className="five"></div>
				<div className="six">
					<div className="bump move-2"></div>
				</div>
				<div className="seven">
					<div className="bump"></div>
				</div>
				<div className="eight">
					<div className="bump move-1"></div>
				</div>
			</div>
		</div>
	</div>
);
export default BB8Loader;
