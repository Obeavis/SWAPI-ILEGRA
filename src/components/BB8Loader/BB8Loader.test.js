import React from 'react';
import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from "react-test-renderer";
import BB8Loader from "./BB8Loader";

configure({ adapter: new Adapter() });

describe('BB8Loader Component', () => {
    const component = (
        <BB8Loader />
    );

    it("matches the snapshot", () => {
        const tree = renderer.create(component).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('Rendering', () => {
        const wrapper = mount(component)
        expect(wrapper.find('.artboard').exists()).toEqual(true)
    });
}); 