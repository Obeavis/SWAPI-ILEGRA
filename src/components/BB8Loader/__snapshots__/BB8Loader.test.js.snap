// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BB8Loader Component matches the snapshot 1`] = `
<div
  className="artboard"
>
  <div
    className="bb8"
  >
    <div
      className="bb8-body"
    >
      <div
        className="dot dot-1"
      >
        <div
          className="line line-1"
        />
        <div
          className="line line-2"
        />
        <div
          className="line line-3"
        />
      </div>
      <div
        className="dot dot-2"
      />
      <div
        className="circle circle-1"
      />
      <div
        className="circle circle-2"
      />
      <div
        className="circle circle-3"
      />
    </div>
    <div
      className="body-shadow-crop"
    >
      <div
        className="body-shadow"
      />
    </div>
    <div
      className="bb8-head"
    >
      <div
        className="head-bottom"
      >
        <div
          className="head-side-1"
        />
        <div
          className="head-side-2"
        />
        <div
          className="head-bottom-base"
        />
      </div>
      <div
        className="head-top-crop"
      >
        <div
          className="head-top"
        />
      </div>
      <div
        className="lens"
      />
      <div
        className="freckle"
      />
    </div>
    <div
      className="speedlines"
    >
      <div
        className="one tail"
      />
      <div
        className="two tail"
      />
      <div
        className="three"
      />
      <div
        className="four"
      />
      <div
        className="five tail"
      />
    </div>
    <div
      className="sparkles"
    >
      <div
        className="one small pulse-1"
      />
      <div
        className="two blue small pulse-2"
      />
      <div
        className="three blue med pulse-3"
      />
      <div
        className="four orange pulse-2"
      />
      <div
        className="five orange pulse-1"
      />
      <div
        className="six blue small pulse"
      />
      <div
        className="seven blue small pulse"
      />
      <div
        className="eight small pulse-3"
      />
      <div
        className="nine pulse"
      />
      <div
        className="ten orange small-1 pulse"
      />
      <div
        className="eleven small pulse"
      />
      <div
        className="twelve small pulse-2"
      />
      <div
        className="thirteen orange small pulse"
      />
      <div
        className="fourteen orange med pulse-3"
      />
      <div
        className="fifteen small pulse-1"
      />
      <div
        className="sixteen small pulse"
      />
    </div>
    <div
      className="ground"
    >
      <div
        className="one"
      >
        <div
          className="bump move-1"
        />
      </div>
      <div
        className="two"
      />
      <div
        className="three"
      >
        <div
          className="bump move-2"
        />
      </div>
      <div
        className="four"
      >
        <div
          className="bump"
        />
      </div>
      <div
        className="five"
      />
      <div
        className="six"
      >
        <div
          className="bump move-2"
        />
      </div>
      <div
        className="seven"
      >
        <div
          className="bump"
        />
      </div>
      <div
        className="eight"
      >
        <div
          className="bump move-1"
        />
      </div>
    </div>
  </div>
</div>
`;
